src/main/java/com/hadoop/compression/lzo/DistributedLzoIndexer.java

```diff
@@ -25,57 +27,108 @@
 
 public class DistributedLzoIndexer extends Configured implements Tool {
   private static final Log LOG = LogFactory.getLog(DistributedLzoIndexer.class);
-  private final String LZO_EXTENSION = new LzopCodec().getDefaultExtension();
 
+  private static final String LZO_EXTENSION = new LzopCodec().getDefaultExtension();
+
+  private static final String LZO_SKIP_INDEXING_SMALL_FILES_KEY = "lzo_skip_indexing_small_files";
+  private static final String LZO_SMALL_FILE_SIZE_KEY = "lzo_small_file_size";
+  private static final String LZO_RECURSIVE_INDEXING_KEY = "lzo_recursive_indexing";
+  private static final boolean LZO_SKIP_INDEXING_SMALL_FILES_DEFAULT = false;
+  private static final boolean LZO_RECURSIVE_INDEXING_DEFAULT = true;
+  private static final long LZO_SMALL_FILE_SIZE_DEFAULT = 0;
+  private boolean lzoSkipIndexingSmallFiles = LZO_SKIP_INDEXING_SMALL_FILES_DEFAULT;
+  private boolean lzoRecursiveIndexing = LZO_RECURSIVE_INDEXING_DEFAULT;
+  private long lzoSmallFileSize = LZO_SMALL_FILE_SIZE_DEFAULT;
+
+  private static final String TEMP_FILE_EXTENSION = "/_temporary";
+
+  private Configuration conf = getConf();
+
+  /**
+   * Accepts paths which don't end in TEMP_FILE_EXTENSION
+   */
   private final PathFilter nonTemporaryFilter = new PathFilter() {
+    @Override
     public boolean accept(Path path) {
-      return !path.toString().endsWith("/_temporary");
+      return !path.toString().endsWith(TEMP_FILE_EXTENSION);
     }
   };
 
-  private void walkPath(Path path, PathFilter pathFilter, List<Path> accumulator) {
+  /**
+   * Returns whether a file should be considered small enough to skip indexing.
+   */
+  private boolean isSmallFile(FileStatus status) {
+    return status.getLen() < lzoSmallFileSize;
+  }
+  
+  private void visitPath(Path path, PathFilter pathFilter, List<Path> accumulator, boolean recursive) {
     try {
-      FileSystem fs = path.getFileSystem(getConf());
+      FileSystem fs = path.getFileSystem(this.conf);
```

Comment:
Nit: the style of this code does not use the "this." notation. Can we remove the "this" reference from the changes?